apps/wallet/src/ui/app/shared/InputWithAction.tsx

```diff
@@ -214,12 +221,30 @@ export const InputWithActionButton = forwardRef<HTMLInputElement, InputWithActio
 						type={type}
 						className={clsx(
 							'bg-transparent z-10 border-none p-0 text-heading5 text-steel-darker font-semibold h-6 caret-hero',
+							loading && 'text-transparent',
 						)}
 						disabled={disabled}
 						ref={forwardRef}
 					/>
+					{loading && (
+						<div className="absolute">
+							<div className="flex items-center gap-1 text-steel">
+								<LoadingIndicator color="inherit" />
+								{loadingText && (
```

Comment:
You don't use this below, it's hardcoded to `Calculating`. Also from what I can tell this is always in a readonly state so it's kind of silly that it's even an input to begin with (don't need to change this just think it's interesting.

Reply:
updated